src/cider/nrepl/middleware/out.clj

```diff
@@ -40,6 +40,32 @@ Please do not inline; they must not be recomputed at runtime."}
             (catch Exception ~'e
               (unsubscribe-session ~'session))))))
 
+(defn- dispatch-string
+  ([messages type ^String x]
+   (.write ^Writer (original-output type) x)
+   (with-out-binding [printer messages type]
```

Comment:
For satisfying kondo, you can teach it about this macro similarly to what is done here:

https://github.com/clojure-emacs/refactor-nrepl/blob/da810357676803d43ab32680b4a244833868c792/.clj-kondo/config.edn#L41-L42